cmd/containerd/command/main.go

```diff
@@ -130,6 +132,18 @@ can be used and modified as necessary as a custom configuration.`
 			return err
 		}
 
+		// Initialize openTelemetry tracing
+		shutdown, err := tracing.InitOpenTelemetry(config)
+		if err != nil {
+			errors.Wrap(err, "failed to initialize OpenTelemetry tracing")
+		}
+		defer shutdown()
+
+		// Get a tracer
+		ctrdTracer := otel.Tracer("containerd")
```

Comment:
So if the config doesn't exist we are effectively disabling OpenTelemetry tracing. Shouldn't the rest of this initialization turn into a no-op at that point? I'll comment at the 2 other places where we have already decided not to initialize. Maybe it has no impact, but if it does (e.g. adding/starting spans, adding gRPC middleware) then we should gate those steps on whether or not it was enabled in the config.

Reply:
That is my current understanding and observations from testing, that all these pieces of code will turn into no-op.